core/router/server.go

```diff
@@ -97,8 +94,8 @@ func (r *routerRPC) Uplink(stream pb.Router_UplinkServer) error {
 		if err != nil {
 			return err
 		}
-		if !uplink.Validate() {
-			return grpcErrf(codes.InvalidArgument, "Invalid Uplink")
+		if err := uplink.Validate(); err != nil {
+			return errors.BuildGRPCError(errors.NewErrInvalidArgument("Uplink", err.Error()))
```

Comment:
Be consistent with `errors.Wrap` vs wrapping it in `errors.NewErrInvalidArgument`